libr/util/str.c

```diff
@@ -1882,6 +1882,53 @@ R_API bool r_str_char_fullwidth (const char* s, int left) {
 
 }
 
+/**
+ * Returns size in bytes of the utf8 char
+ * Returns 1 in case of ASCII
+ * str - Pointer to buffer
+ */
+R_API int r_str_utf8_charsize(const char *str) {
+	r_return_val_if_fail (str, 0);
+	int size = 0;
+	int length = strlen (str);
+	while (size < length) {
+		size++;
+		if ((str[size] & 0xc0) != 0x80) {
+			break;
+		}
+	}
+	return size;
+}
+
+/**
+ * Returns size in bytes of the previous utf8 char
+ * Returns 1 in case of ASCII
+ * str - Pointer to leading utf8 char
+ * len - Length in bytes of the buffer previous to str
+ */
+R_API int r_str_utf8_charsize_reverse(const char *str, int len) {
```

Comment:
i would handle if len < 0 perform strlen

Reply:
but... whats the point on having a reverse charsize here?

Reply:
len is inverse len, which is kind of confusing and errorprone

Reply:
the reverse charsize is for knowing how many bytes to the left you need to move the index so that it is always on the leading utf8 byte